app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,50 @@
+class BookingsController < ApplicationController
+  def index
+    @bookings = Bookings.all
+  end
+
+  def new
+    @listing = Listing.find(params[:listing_id])
+    @booking = Booking.new
+  end
+
+  def create
+    @booking = Booking.new(booking_params)
+    @listing = Listing.find(params[:listing_id])
+    @booking.listing = @listing
+    if @booking.save
+      redirect_to booking_path(@booking), notice: 'Booking was successfully created!'
```

Comment:
The route name will be `listing_booking_path`

Reply:
I made the requested change.